src/ets/scala/io/iohk/ethereum/ets/blockchain/ScenarioSetup.scala

```diff
@@ -19,7 +20,11 @@ abstract class ScenarioSetup(vm: VMImpl, scenario: BlockchainScenario)
   with ValidatorsBuilder
   with SyncConfigBuilder
   with BlockchainConfigBuilder
-  with ActorSystemBuilder {
+  with ActorSystemBuilder
+  with ConsensusBuilder
+  with ConsensusConfigBuilder
+  with ShutdownHookBuilder
+  with Logger {
```

Comment:
Why were all of these added?

Reply:
Let me recheck this. In the next PR, I have a new trait that just mixes in everything you need to get a `consensus` so as to not write all these `with` every time but I am not sure it applies here.

Reply:
It's complicated. `Validators` are now (as in the next PR) the responsibility of `Consensus`, so the source of truth for them is consensus, they are not a standalone entity. They do not actually use any other consensus functionality but they are **part** of it. In fact `ValidatorsBuilder` [reduces to](https://github.com/input-output-hk/mantis/blob/4b4c76d5cffe4f28ef696bff4ec9fb2e3fbc3405/src/main/scala/io/iohk/ethereum/nodebuilder/NodeBuilder.scala#L352..L356):

```scala
trait ValidatorsBuilder {
  self: ConsensusBuilder =>

  lazy val validators = consensus.validators
}
```

Note: the way we stack and mix traits, and use them especially in the tests, is a source of complexity that makes it very difficult to cope with all these changes. A tiny refactor somewhere (e.g. in `Ledger`) looks like an amplification attack to the whole codebase :)